File: 2025/cpp/junior/junior3.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int n;
  cin >> n;

  for (int i = 0; i < n; i++) {
    string s;
    cin >> s;
    int total = 0, current = 0;
    string uppercases = "";
    bool negative = false;

    for (auto c : s) {
      if (isalpha(c)) {
        if (negative) {
          total -= current;
        } else {
          total += current;
        }
        current = 0;
        negative = false;
        if (isupper(c)) {
          uppercases += c;
        }
      } else if (c == '-') {
        if (negative) {
          total -= current;
        } else {
          total += current;
        }
        current = 0;
        negative = true;
      } else {
        current = current * 10 + (c - '0');
      }
    }
    if (negative) {
      total -= current;
    } else {
      total += current;
    }

    cout << uppercases << total << endl;
  }
}

File: 2025/python/junior/junior3.py
n = int(input())

for _ in range(n):
    s = input()
    total = 0
    current = 0
    uppercases = ""
    negative = False
    for c in s:
        if c.isalpha():
            if negative:
                total -= current
            else:
                total += current
            current = 0
            negative = False
            if c.isupper():
                uppercases += c
        elif c == '-':
            if negative:
                total -= current
            else:
                total += current
            current = 0
            negative = True
        elif c.isdigit():
            current = current * 10 + int(c)
    if negative:
        total -= current
    else:
        total += current
    print(uppercases + str(total))
